packages/vitest/src/integrations/wait.ts

```diff
@@ -95,3 +95,82 @@ export function waitFor<T>(callback: WaitForCallback<T>, options: number | WaitF
     intervalId = setInterval(checkCallback, interval)
   })
 }
+
+export type WaitUntilCallback = () => boolean | Promise<boolean>
```

Comment:
I don't think we need to ask for a boolean. Truthy values are sufficient. It would also be nice to actually return the value:

```ts
const element = await vi.waitUntil(() => document.querySelector('.element'))
```

Reply:
Does waitFor also return the value? Would be nice to support it in both methods

Reply:
> Does waitFor also return the value? Would be nice to support it in both methods

The `waitFor` already supports the return callback's value